### PR TITLE
mediathekview: 9 -> 13.2.1

### PR DESCRIPTION
**Pull request by Moredread**
###### Motivation for this change

First, `zdfmediathk` seems to be a duplicate of `mediathekview`, as the upstream project is https://mediathekview.de now. (Not yet removed in this PR).

I got the recent version almost working, but the dependency `OpenJFX` isn't available in nix yet. See issue #16727 for that.

I used the `zdfmediathk` expression as the base for the update as it seemed more complete, but the package name `mediathekview` matches how upstream calls the package.

@chaoflow @flosse: Are you still interested in maintaining MediathekView? Have you tried updating the package recently?

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/video/mediathekview/default.nix [hunk 1]
```diff
@@ -1,31 +1,39 @@
...
+    install -m644 -t $out/lib lib/*
+    install -m755 bin/flv.sh $out/bin/
+    install -m644 -t $out/share/doc Anleitung/*.pdf
+    cd Copyright
+    find -type f -exec install -Dm 755 "{}" "$out/share/licenses/{}" \;
+    cd ..
+    bin="$out/bin/mediathek"
+    cat >> "$bin" << EOF
+    #!${bash}/bin/sh
+    exec ${jre}/bin/java -cp "$out/lib/*" -Xms128M -Xmx1G -jar "$out/MediathekView.jar" "\$@"
```
Comment (Mic92) [inline pkgs/applications/video/mediathekview/default.nix]: use makeWrapper here.

**Comment by Mic92** [comment]
I get now:

```connsole
$ mediathek
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=lcd -Dswing.defaultlaf=com.sun.java.swing.plaf.gtk.GTKLookAndFeel
===========================================
JavaFX wurde nicht im klassenpfad gefunden.
 Stellen Sie sicher, dass Sie ein Java JRE ab Version 8 benutzen.
 Falls Sie Linux nutzen, installieren Sie das openjfx-Paket ihres Package-Managers,
 oder nutzen Sie eine eigene JRE-Installation.
===========================================
```

**Comment by Mic92** [comment]
Ah also mentioned in the description.

**Comment by Moredread** [comment]
Btw, the old version isn't working anymore. If you want to use it in the meantime you can use it with the Oracle JRE, as it has JavaFX bundled.

That might be a solution for the package as well until openjfx is bundled.

**Comment by Moredread** [comment]
I've updated the PR so it now uses the OracleJRE. As the package isn't usable at the moment otherwise, I think this is the best way to go forward until OpenJFX is supported.

**Comment by Moredread** [comment]
@Mic92 Could you have a look again please?

**Comment by Mic92** [comment]
There is still an evaluation error for darwin, which I try to fix in:
https://github.com/NixOS/nixpkgs/pull/54044/files

**Comment by Moredread** [comment]
@Mic92 I've cleaned up a few things (derivation inputs, removed java flags) and removed the obsolete `zdfmediathk` package as well.

File: pkgs/applications/video/mediathekview/default.nix [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+    url = "https://download.mediathekview.de/stabil/MediathekView-${version}.tar.gz";
+    sha256 = "11wg6klviig0h7pprfaygamsgqr7drqra2s4yxgfak6665033l2a";
   };
-  unpackPhase = "true";
 
-  buildInputs = [ unzip ];
-  
-  # Could use some more love
-  # Maybe we can also preconfigure locations for vlc and the others.
+  buildInputs = [ makeWrapper ];
```
Comment (dotlambda) [inline pkgs/applications/video/mediathekview/default.nix]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/applications/video/mediathekview/default.nix [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
   meta = with stdenv.lib; {
-    homepage = http://zdfmediathk.sourceforge.net/;
-    license = stdenv.lib.licenses.gpl3;
-    maintainers = [ maintainers.chaoflow ];
-    platforms = platforms.linux;  #  also macOS and cygwin, but not investigated, yet
+    description = "Offers access to the Mediathek of different tv stations (ARD, ZDF, Arte, etc.)";
+    homepage = https://mediathekview.de/;
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ chaoflow moredread ];
+    platforms = platforms.all;
```
Comment (dotlambda) [inline pkgs/applications/video/mediathekview/default.nix]: Did you test this?

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/applications/video/mediathekview/default.nix (hunk 3 above)
Comment (Mic92) [reply pkgs/applications/video/mediathekview/default.nix]: The limiting factor here would be probably our bash wrapper that does not run as it is on windows.

**Comment by Mic92** [comment]
@GrahamcOfBorg build mediathekview

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -23077,8 +23079,6 @@ in
 
   zap = callPackage ../tools/networking/zap { };
 
-  zdfmediathk = callPackage ../applications/video/zdfmediathk { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Can you add an alias in pkgs/top-level/aliases.nix?

File: pkgs/applications/video/mediathekview/default.nix (hunk 3 above)
Comment (Moredread) [reply pkgs/applications/video/mediathekview/default.nix]: I took it from one of the old `zdfmediathk` package. Should I change it or leave it as is?

**Pull request merged**